plugin/trino-kinesis/src/main/java/io/trino/plugin/kinesis/KinesisRecordSet.java

```diff
@@ -290,6 +300,32 @@ private void getKinesisRecords()
             messagesRead += kinesisRecords.size();
         }
 
+        private Optional<Map<DecoderColumnHandle, FieldValueProvider>> decodeMessage(byte[] messageData)
+        {
+            if (!split.getCompressionCodec().isPresent()) {
+                return messageDecoder.decodeRow(messageData);
+            }
+            if (split.getCompressionCodec().get() == GZIP || split.getCompressionCodec().get() == AUTOMATIC) {
```

Comment:
Extract variable for `split.getCompressionCodec().get()`. Or even introduce a method to enum `canUseGzip()` which would return true for `AUTOMATIC` and `GZIP`